embrace-android-delivery/src/main/kotlin/io/embrace/android/embracesdk/internal/delivery/scheduling/SchedulingServiceImpl.kt

```diff
@@ -33,8 +33,7 @@ class SchedulingServiceImpl(
     private val sendLoopActive = AtomicBoolean(false)
     private val queryForPayloads = AtomicBoolean(true)
     private val activeSends: MutableSet<StoredTelemetryMetadata> = Collections.newSetFromMap(ConcurrentHashMap())
-    private val completedSendsSinceQueueDrain: MutableSet<StoredTelemetryMetadata> =
-        Collections.newSetFromMap(ConcurrentHashMap())
+    private val deleteInProgress: MutableSet<StoredTelemetryMetadata> = Collections.newSetFromMap(ConcurrentHashMap())
     private val payloadsToRetry: MutableMap<StoredTelemetryMetadata, RetryInstance> = ConcurrentHashMap()
```

Comment:
This isn't essential for merging this PR or shipping, but I do wonder whether we should formalise these states with an enum and store the status in one map. That avoids the possibility that statuses show up in 0 or >1 collection at the same time & likely simplifies the code.

Reply:
That's a great idea! It also opens up the possibly for a 3rd state - it's completed, but we are liable to change it because reasons. I wouldn't mind doing it now given it should be pretty isolated.

Now as in in a PR after this is merged 😅